pkg/kubectl/cmd/util/factory.go

```diff
@@ -178,6 +178,9 @@ type Factory interface {
 	PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool) (kubectl.ResourcePrinter, error)
 	// One stop shopping for a Builder
 	NewBuilder() *resource.Builder
+
+	// SuggestPodTemplateResources returns a list of resource types that declare a pod template
+	SuggestPodTemplateResources() []unversioned.GroupResource
```

Comment:
nit: `SuggestedPodTemplateResources` since this is returning a noun, rather than performing an action.